opengever/maintenance/scripts/fix_retention_period.py

```diff
@@ -0,0 +1,465 @@
+
+        parent_context = aq_parent(aq_inner(self.context))
+        self.parent = registry['/'.join(parent_context.getPhysicalPath())]
+        self.parent.append_child_folder(self)
+
+    @property
+    def is_leaf_folder(self):
+        if self._is_leaf_folder is None:
+            child_folders = self.catalog.unrestrictedSearchResults(
+                path={'query': '/'.join(self.context.getPhysicalPath()),
+                      'depth': 1},
+                object_provides=IRepositoryFolder.__identifier__)
+            self._is_leaf_folder = len(child_folders) == 0
+        return self._is_leaf_folder
+
+    def _diff(self):
+        self.reference_number = self._get_repository_reference_number()
+        if self.reference_number != self.item['reference_number']:
+            logger.warn('reference numbers differ for: {}'
+                        ' "{}" (site), "{}" (excel)'
+                        .format(self.item['_query_path'],
+                                self.reference_number,
+                                self.item["reference_number"]))
+            self.can_apply = False
+            return
+
+        self.current_period = ILifeCycle(self.context).retention_period
+        xls_period = self.item.get('retention_period')
+        self.new_period = xls_period or self.parent.new_period
+
+    def _get_repository_reference_number(self):
+        reference = IReferenceNumber(self.context)
+        formatter = getAdapter(self.context, IReferenceNumberFormatter,
+                               name=self.reference_formatter)
+        return formatter.repository_number(reference.get_parent_numbers())
+
+    def make_retention_period_backup(self, obj, backup_period):
+        annotations = IAnnotations(obj)
+        if BACKUP_KEY not in annotations:
+            annotations[BACKUP_KEY] = backup_period
+
+    def apply_recursively(self):
+        if not self.can_apply:
+            return
+
+        if self.apply_to_repo_folder() and self.is_leaf_folder:
+            self.apply_to_dossiers()
+
+        super(RepoFolderDiff, self).apply_recursively()
+
+    def apply_to_dossiers(self):
+        child_dossiers = self.catalog.unrestrictedSearchResults(
+            path='/'.join(self.context.getPhysicalPath()),
+            object_provides=IDossierMarker.__identifier__)
+
+        for brain in child_dossiers:
+            self.apply_to_dossier(brain.getObject())
+
+    def apply_to_repo_folder(self):
+        kind = 'leaf ' if self.is_leaf_folder else ''
+        if self.current_period != DEFAULT_PERIOD:
+            if self.options.verbose:
+                logger.info('skipping {}repo-folder ({}) {}, modified'
+                            .format(kind,
+                                    self.reference_number,
+                                    self.item['_query_path']))
+            return False
+
+        current_title = self.context.Title(prefix_with_reference_number=False)
```

Comment:
yes, as discussed the french title is currently ignored here since german is default. this should be good enough™️.